src/saveload/afterload.cpp

```diff
@@ -2763,6 +2763,36 @@ bool AfterLoadGame()
 		}
 	}
 
+	if (IsSavegameVersionBefore(SLV_DEPOTID_IN_HANGAR_ORDERS)) {
+		/* Update go to hangar orders so they store the DepotID instead of StationID. */
+		for (Aircraft *a : Aircraft::Iterate()) {
+			if (!a->IsNormalAircraft()) continue;
+			if (a->orders == nullptr) continue;
+			if (a->orders->GetFirstSharedVehicle() != a) continue;
+
+			for (Order *order : a->Orders()) {
+				if (!order->IsType(OT_GOTO_DEPOT)) continue;
+				StationID station_id = order->GetDestination();
+				assert(Station::IsValidID(station_id));
+				Station *st = Station::Get(station_id);
+				assert(st->airport.hangar != nullptr);
+				order->SetDestination(st->airport.hangar->index);
+			}
+		}
+
+		for (Aircraft *a : Aircraft::Iterate()) {
+			if (!a->IsNormalAircraft()) continue;
+			if (!a->current_order.IsType(OT_GOTO_DEPOT)) continue;
+
+			Station *st = Station::Get(a->current_order.GetDestination());
+			if (st->airport.hangar != nullptr) {
```

Comment:
The loop above asserts that `st->airport.hangar != nullptr`. Is it possible here?

Reply:
Technically, it shouldn't be possible. But I am not sure all savegames are bug free and I know there have been some bugs related to hangar orders in the past. I will change the "if" with an assert, just in case.

I could remove the assert too, but I would keep it just in case.

Reply:
Well, they should match, at least. Keep in mind that assertions are used for development and testing only and are disabled in release builds. So if you expect users to load corrupted savegames, an assertion won't prevent this.